syntax = "proto3";
// Value Objects


message ProjectMetadataDelta {
    string name = 1;
    bool name_set = 2;

    string description = 3;
    bool description_set = 4;
}




message ServiceMetadataDelta {
    string service_name = 1;
    bool service_name_set = 2;
    string location_id = 3;
    bool location_id_set = 4;

    string location_uri = 5;
    bool location_uri_set = 6;

    string description = 7;
    bool description_set = 8;

    repeated string experts = 9;
    bool experts_set = 10;
}



// if a field kind is set, then the corresponding property carries
// a new value (which could be null)
message DatasetMetadataDelta {



    int64 record_count = 1;
    bool record_count_set = 2;
    int64 file_count = 3;
    bool file_count_set = 4;
    int64 storage_bytes = 5;
    bool storage_bytes_set = 6;
    DatasetSample sample = 7;
    bool sample_set = 8;

    int64 update_timestamp = 9;
    bool update_timestamp_set = 10;

    string data_format = 11;
    bool data_format_set = 12;
    string description = 13;
    bool description_set = 14;

    string location_id = 15;
    bool location_id_set = 16;
    string location_uri = 17;
    bool location_uri_set=18;


    repeated string experts = 19;
    bool experts_set = 20;

    string name = 21;
    bool name_set = 22;
}





message DatasetSample {
    // Sample storage format. Doesn't necessarily match
    // actual data format which could be very cryptic
    enum FORMAT {
        TEXT = 0;
        TSV = 1;
        JSON = 2;
    }

    FORMAT format = 1;
    string body = 2;
}

